kuma/wiki/models.py

```diff
@@ -811,6 +809,9 @@ def revert(self, revision, user, comment=None):
         revision.created = datetime.now()
         revision.creator = user
         revision.save()
+        if revision.document.original == self:
+            revision.based_on = revision
+        revision.save()
```

Comment:
Hmm ... line 811 above it calls `save()` and triggers the `post_save` callbacks already, so it's the same order of operations we have now. But a double-call to `save()` and double `post_save` signals does seem risky. Hmm ...